module-2/lab-hypothesis-testing/your-code/challenge-1.ipynb

```diff
@@ -19,13 +19,16 @@
   },
```

Comment:
- Muy bien por incorporar el supuesto de no igualdad de varianzas, yo haría también varianzas iguales. La verdad es que este lab está sujeto a la hipótesis de partida sobre si consideramos dos muestras de una misma población o dos muestras de dos poblaciones diferentes. Si asumimos que el dataset es la población de pokemons y seleccionamos dos muestras de esta población, podríamos correr un paired t-test. 
- Fenomenal generar las muestras de forma aleatoria y de mismo tamaño así no tienes que preocuparte de la homogeneidad de las varianzas. En el mundo real podríamos tener bastantes diferencias de tamaño entre dos muestras. Muchos piensan que es mejor usar Welch’s t-test porque desempeña mejor que el t-test tanto en varianzas iguales como en descalces importantes en tamaño de las muestras. Ya sabes,en estadística nunca 2+2 son 4. 
- Hay veces que la notación científica nos puede dar lugar a quebraderos de cabeza. Para las calculadoras está bien, pero el cerebro humano funciona mejor con float. Suele ser útil cambiar el formato a float. Hay varias formas de hacer esto en python, te escribo dos para que las puedas aplicar cuando lo consideres: 
```Python
format(valor, 'f')
format(valor, '.<número de decimales>f')
```
- Por notación estándar, rechazamos hipótesis nula (reject H0) o no podemos rechazar la hipótesis nula (fail to reject).
- Puedes echarle un vistazo a ésto? Creo que el check contra el nivel de significación está invertido. 
```Python
    testt=stats.ttest_rel(s1,s2)[1]
    list_of_SASD.append('H0 refuted' if testt>0.05 else 'H0 confirmed')
    print('Attack vs Defense: ',testt,' >> ','H0 refuted' if testt>0.05 else 'H0 confirmed')
```

- Al hacer [1] entiendo que sólo estás sacando el p-value del test. Es útil en los contrastes de hipótesis imprimir todos los resultados, no sólo el p-value. En este caso el estadístico t. 
```Python
Attack vs defense Tests: Refuted  0 confirmed  10
  >>  There are significant differences
Sp. Atk vs Sp. Def Tests: Refuted  10 confirmed  0
  >>  No differences
```
Es al revés, al estar cambiado el check. 
Una forma muy sencilla y legible de hacerlo sería
```Python
print(stats.ttest_rel(pokemon.Attack, pokemon.Defense))
print(stats.ttest_rel(pokemon['Sp. Atk'], pokemon['Sp. Def']))
```
- En términos generales, el poder de nuestros test de contraste está muy relacionado con el tamaño de la muestra. En estos casos de número bajo de observaciones, el poder es mucho menor y aún obteniendo p-values que nos llevaran a no rechazar la hipótesis nula, tendríamos que hacer contrastes adicionales antes de aceptar sin más nuestra hipótesis de igualdad de medias.